vignettes/data-transform-as-shiny-module.Rmd

```diff
@@ -1,26 +1,30 @@
 ---
-title: "Data Transform as shiny Module"
+title: "Data Transformations as shiny Module"
 author: "NEST CoreDev"
 output: 
   rmarkdown::html_vignette:
     toc: true
 vignette: >
-  %\VignetteIndexEntry{Data Transform as shiny Module}
+  %\VignetteIndexEntry{Data Transformations as shiny Module}
   %\VignetteEngine{knitr::rmarkdown}
   %\VignetteEncoding{UTF-8}
 ---
 
 ## Introduction
 
-When data transformations need to occur inside the teal app, they typically happen within the teal module development by the teal module developer. This results in the transformation code being tightly coupled with the teal module. Any changes to the data transformation logic require updating the teal module, which can impact the teal app development until the module is updated.
-With the new `teal_transform_module()` function, it is possible to provide a more dynamic data transformation capability that can be customized by the teal app developer, not just the teal module developer.
-The `teal_transform_module()` function is essentially a Shiny module that takes `ui` and `server` arguments and can be provided to the `transformer` argument in a `module()`. When provided, teal will execute the data transformation logic for the specified module when it is loaded. 
-In addition, this approach makes it easier for teal app developers to handle additional data transformations while using the teal modules R packages like `teal.modules.general` and `teal.modules.clinical`.
-Let’s explore this new way to manage custom data transformations in teal apps.
+`teal` ver `0.15.x` (TO BE UDPATED) introduced new argument in `teal::module` called `transformers`. 
+This argument allows to pass a `list` of `teal_data_module` objects that are created either with `teal_transform_module()` or `teal_data_module()` functions.
```

Comment:
I'm unsure if it's necessary to mention using the `teal_data_module()` function to create the object and pass it to the `transformers` argument. This may make sense from a developer's perspective since `teal_transform_module()` also returns a `teal_data_module` class, but I'm not sure if it's useful information for the user.

In all the examples below, we only use `teal_transform_module()` to create the object passed to `transformers`.

If we decide to keep the mention of `teal_data_module()`, we would need to include examples of how to pass the object to `transformers` using that function. I expect that using `teal_data_module()` would involve additional steps, which might not be ideal since `teal_transform_module()` already handles this.

At the moment, I prefer to focus on `teal_transform_module()`, but I'd like to hear your thoughts on this.

```suggestion
This argument allows to pass a `list` of `teal_data_module` objects that are created using `teal_transform_module()` function.
```

Reply:
Agree. I also had those thoughts when I wrote this sentence. Will accept the change

Reply:
Actually @vedhav can you accept during your last review? This all is ready to be accepted and merged

Reply:
Good catch Dony, It's actually only important to talk about the object `teal_data_module` in the context of the transform module to make sure that we communicate that both the DDL and transform use the same object.
But, right now we already have different features between the object created with `teal_transform_module()` and `teal_data_module()` that it does not make sense to use the later for transforming.